### ISSUE TITLE
Slow scan

### ISSUE THREAD

**Issue by jbhunter21** [khaytsus/gqrx-scan#14]
Trying to do a test scan through 127Mhz - 128Mhz with this command:
gqrx-scan --type scan --start 127000000 --end 128000000 --mode AM --delaylevel=-50 --delaytime 10

Problem is that gqrx-scan changes the frequency really slow. It changes about every 5th second instead of the default 0.5 second interval.

I have installed all Perl-modules specified in the readme file.

**Comment by khaytsus** [comment]
I've seen very slow scanning before when I had direct sampling enabled.  Have you by chance played with that in the drivers and/or set the option in GQRX?  I played with direct sampling at one point and was pretty disappointed with the reception but was more disappointed that my scanning slowed way down.

If not, have you played with any settings in the script, are you using a ~/gqrx-scan-defaults.pm file ?  If so, could you let me know what settings you've changed, perhaps paste the defaults config file if you've modified it?  In particular, if the $pause has been increased this would directly affect scan speed.  If it's been decreased, the script will only allow it to go as low as .3s as anything faster is unreliable when checking the signal level.

**Comment by jbhunter21** [comment]
I'm using the ~/gqrx-scan-defaults.pm and have not changed any parameters within it. One thing that might have been the cause is I happend to have gqrx 2.2 installed and not 2.3 or above as required.

Now I have installed gqrx 2.4 but having a new issue.

./gqrx-scan --type scan --start 127000000 --end 128000000 --mode AM --delaylevel=-50 --delaytime 10
Loading default config file /home/rambo/gqrx-scan-defaults.pm

Starting in scan mode from 127.000 000 to 128.000 000 in AM mode
Signal level delay level of -50 waiting for 10s

Channel/Frequency pause time:  0.5s
Delaylevel: [-50]  Delaytime: [10]

write error: filehandle isn't open at ./gqrx-scan line 494

**Comment by khaytsus** [comment]
You can't use GQRX 2.2; it doesn't have the TCP remote control.  So you at least have 2.3, or possibly something that's 2.2ish but has some 2.3 features?  Maybe some intermediate version?  Not sure there, but pretty sure 2.2 wouldn't have worked at all.  :)

Odd that you'd get a filehandle error that far into the code....  If you didn't have the TCP remote enabled it'd fail much earlier.  I'm not on 2.4 (just noticed it was released) but I'm on a very recent git build, so I can't imagine anything has changed here but I'll pull the latest build in a while and check the script against it.  In the meanwhile, just in case I missed some error handling, make sure TCP remote is enabled in GQRX and your script and/or config are pointing to the right IP and port.

**Comment by jbhunter21** [comment]
Got it to work by removing the IPv6 address that where in the TCP settings of gqrx and leaving only 127.0.0.1 left.
Now everything is working as intended :smile:

**Comment by khaytsus** [comment]
Awesome!  I'm still confused about your 2.2 usage, but hey, it's working now.  I'll make a note of that in the readme about ipv6.  Thanks!

**Comment by jbhunter21** [comment]
To make it a bit more clear. My scan-speed problem was due to version 2.2 I believe. After I upgraded gqrx to version 2.4 I got the filehandle problem, solving it by removing the IPv6 address that was default configured under the TCP setting sin gqrx version 2.4.

**Comment by khaytsus** [comment]
But 2.2 didn't have the TCP Remote control; so the script couldn't have worked at all :)  But all is well now.  :)

**Comment by khaytsus** [comment]
I ran into this too in 2.5, apparently it by default adds ipv6 ::1 but not localhost?  I might file a bug in GQRX for that, but I took it as an opportunity to add better connection error handling to the script.  Now I detect when we can't connect or we're not getting sane data back.  There might be other cases I've missed, I might think about this some more.  Added in 3999897ab7395147ce62cb8ffe09716c0ced82d2
